src/include/parsing/utils/parsing_table.h

```diff
@@ -0,0 +1,147 @@
+#ifndef JUCC_PARSING_UTILS_PARSING_TABLE_H
+#define JUCC_PARSING_UTILS_PARSING_TABLE_H
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
```

Comment:
remove unnecessary headers from .h. Add in .cpp if required. keeps the header file lighter.